Common/Securities/SecurityDefinitionSymbolResolver.cs

```diff
@@ -0,0 +1,163 @@
+        /// to get the ticker of the symbol on this date.
+        /// </param>
+        /// <returns>The composite FIGI's corresponding Symbol</returns>
+        public Symbol CompositeFIGI(string compositeFigi, DateTime tradingDate)
+        {
+            return SecurityDefinitionToSymbol(
+                _securityDefinitions?.FirstOrDefault(x => x.FIGI == compositeFigi && !string.IsNullOrWhiteSpace(x.FIGI)),
+                tradingDate);
+        }
+        
+        /// <summary>
+        /// Converts SEDOL into a Lean <see cref="Symbol"/>
+        /// </summary>
+        /// <param name="sedol">SEDOL</param>
+        /// <param name="tradingDate">
+        /// The date that the stock was trading at with the SEDOL provided. This is used
+        /// to get the ticker of the symbol on this date.
+        /// </param>
+        /// <returns>The SEDOL's corresponding Symbol</returns>
+        public Symbol SEDOL(string sedol, DateTime tradingDate)
+        {
+            return SecurityDefinitionToSymbol(
+                _securityDefinitions?.FirstOrDefault(x => x.SEDOL == sedol && !string.IsNullOrWhiteSpace(x.SEDOL)),
+                tradingDate);
+        }
+
+        /// <summary>
+        /// Converts ISIN into a Lean <see cref="Symbol"/>
+        /// </summary>
+        /// <param name="isin">ISIN</param>
+        /// <param name="tradingDate">
+        /// The date that the stock was trading at with the ISIN provided. This is used
+        /// to get the ticker of the symbol on this date.
+        /// </param>
+        /// <returns>The ISIN's corresponding Lean Symbol</returns>
+        public Symbol ISIN(string isin, DateTime tradingDate)
+        {
+            return SecurityDefinitionToSymbol(
+                _securityDefinitions?.FirstOrDefault(x => x.ISIN == isin),
+                tradingDate);
+        }
+
+        /// <summary>
+        /// Converts a SecurityDefinition to a <see cref="Symbol" />
+        /// </summary>
+        /// <param name="securityDefinition">Security definition</param>
+        /// <param name="tradingDate">
+        /// The date that the stock was being traded. This is used to resolve
+        /// the ticker that the stock was trading under on this date.
+        /// </param>
+        /// <returns>Symbol if matching Lean Symbol was found on the trading date, null otherwise</returns>
+        private Symbol SecurityDefinitionToSymbol(SecurityDefinition securityDefinition, DateTime tradingDate)
+        {
+            if (securityDefinition == null)
+            {
+                return null;
+            }
+            
+            // Get the first ticker the symbol traded under, and then lookup the
+            // trading date to get the ticker on the trading date.
+            var mappedTicker = _mapFileResolver
+                .ResolveMapFile(securityDefinition.SecurityIdentifier.Symbol, securityDefinition.SecurityIdentifier.Date)
+                .GetMappedSymbol(tradingDate);
```

Comment:
`GetMappedSymbol` wont map if `tradingDate` is past the delisting date, maybe to be more user friendly should just return the symbol unmapped?

Reply:
Making it so that we return the last known symbol now :+1: